internal/logging/filtering.go

```diff
@@ -60,19 +60,39 @@ func (lo LoggerOpts) ApplyMask(msg *string, fieldMaps ...map[string]interface{})
 		}
 	}
 
-	// Replace any part of the log message matching any of the configured regexp,
-	// with a masking replacement string
 	if len(lo.MaskMessageRegexes) > 0 {
 		for _, r := range lo.MaskMessageRegexes {
+			// Replace any part of the log message matching any of the configured regexp
 			*msg = r.ReplaceAllString(*msg, logMaskingReplacementString)
+
+			// Replace any part of any log field matching any of the configured regexp
+			for _, f := range fieldMaps {
+				for fk, fv := range f {
+					// Can apply the regexp replacement, only if the field value is indeed a string
+					fvStr, ok := fv.(string)
```

Comment:
However this functionality lands, we'll likely need to update at least the Go documentation for this particular limitation. Maybe in the future it can be fancier about handling `[]byte` or anything we can find strings in via reflection.

Reply:
Good call out about documenting it.

I did think a bit about what we could do there, but it feels rather tricky to handle all the possible cases, rather than just impose a limitation. It's not unreasonable to expect that "if you want this kind of filtering done to your field values, you have to use a string".

Reply:
Good call out about documenting it.

I did think a bit about what we could do there, but it feels rather tricky to handle all the possible cases, rather than just impose a limitation. It's not unreasonable to expect that "if you want this kind of filtering done to your field values, you have to use a string".

Reply:
I think I have addressed everything you raised.

* separate functions to configure masking for the content of the value of the fields alone
* document the fact that can only work when the field value is an actual string
* additional `MaskLogRegexes` and `MaskLogStrings` that internally call the functions specific for fields and message (the doc reflects that)
* added website documentation
* updated the changelog entry accordingly